### DISCUSSION TITLE
default constraint syntax `...value` not working

### DISCUSSION THREAD

**Discussion by heistp** [cue-lang/cue#1529] [Q&A] [answered]
In the [language spec Structs section](https://cuelang.org/docs/references/spec/#structs), it states that a default constraint may be specified with `...value`, which should declare that all fields with no other declaration are unified with `value`.

However, when using CUE v0.4.2, while the following works:

```
#Props: {
	...
}

p: #Props & {
	a: "1"
	b: "2"
}
$ cue export props.cue                                                
{
    "p": {
        "a": "1",
        "b": "2"
    }
}
```

Adding a value to `...` does not:

```
#Props: {
	...string
}

p: #Props & {
	a: "1"
	b: "2"
}
$ cue export props.cue                                                
missing ',' in struct literal:
    ./props.cue:2:5
expected '}', found 'EOF':
    ./props.cue:8:3
```

It seems like either the language spec or the implementation must be wrong here, because the example in the language spec does not work either:
```
a: {
    foo:    string    // foo is a string
    [=~"^i"]: int     // all other fields starting with i are integers
    [=~"^b"]: bool    // all other fields starting with b are booleans
    ...string         // all other fields must be a string
}

b: a & {
    i3:    3
    bar:   true
    other: "a string"
}
$ cue export example.cue                                              
missing ',' in struct literal:
    ./example.cue:5:8
expected '}', found 'EOF':
    ./example.cue:12:3
```

Is there an updated syntax for this that needs documentation, or is this a bug?

**Comment by slewiskelly** [comment, marked as answer]
https://github.com/cue-lang/cue/pull/1471 addresses this, as it has yet to be implemented.

**Reply by heistp** [reply]
Thanks!
